src/graphnet/data/dataset.py

```diff
@@ -283,6 +310,16 @@ def _create_graph(
         graph = Data(x=x, edge_index=None)
         graph.n_pulses = n_pulses
         graph.features = self._features[1:]
+        # Write add loss weight to graph.
+        if loss_weight is not None and self._loss_weight_column is not None:
+            if len(loss_weight) == 0:
+                graph[self._loss_weight_column] = torch.tensor(
+                    self._loss_weight_padding_value, dtype=self._dtype
+                ).reshape(-1, 1)
```

Comment:
I am not sure what this guards against, if `loss_weight` is either None or a float, see the type hints and docstring?